source/MaterialXCore/Element.h

```diff
@@ -71,6 +71,92 @@ using ElementMap = std::unordered_map<string, ElementPtr>;
 /// A standard function taking an ElementPtr and returning a boolean.
 using ElementPredicate = std::function<bool(ConstElementPtr)>;
 
+/// @class ElemenEquivalenceResult
+/// The results of comparing for equivalence.
+class MX_CORE_API ElementEquivalenceResult
```

Comment:
Ideally we should declare these classes _after_ the core `Element` class, since a new developer reading through our code needs to understand the `Element` class before they can understand these.

Since `ElementEquivalenceResult` objects are always returned via pointer, it should be possible to forward declare this class in the header, and it may be worthwhile to find a way to forward declare `ElementEquivalenceOptions` as well.